### PR TITLE
feat(custom-roles): added support to import/export custom-roles

### PR DESCRIPTION
**Pull request by vkalta**
This relates to CS-27152

### CODE REVIEW

File: packages/contentstack-import/src/lib/import/custom-roles.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    if (!self.customRoles) {
+      addlogs(config, chalk.white('No custom-roles found'), 'error');
+      return;
+    }
+    self.customRolesUids = Object.keys(self.customRoles);
+    for (const uid of self.customRolesUids) {
+      const customRole = self.customRoles[uid];
+
+      if (uid in self.customRolesUidMapper) {
```
Comment (shafeeqd959) [inline packages/contentstack-import/src/lib/import/custom-roles.js:52]: do we to loop in mapper to check it already created or not, or  self.customRolesUidMapper[uid]  o(1) operation

File: packages/contentstack-export/src/config/default.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ module.exports = {
     types: [
       'stack',
       'assets',
+      'custom-roles',
```
Comment (shafeeqd959) [inline packages/contentstack-export/src/config/default.js:12]: keep the custom roles just above the workflows, is there any other module dependent on custom roles ?

File: packages/contentstack-import/src/config/default.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ module.exports = {
   modules: {
     types: [
       'locales',
+      'custom-roles',
```
Comment (shafeeqd959) [inline packages/contentstack-import/src/config/default.js:14]: Keep it just above the workflows , is there any other module other than workflow dependent on the custom roles ? is there any reason that you kept custom roles after locales

**Comment by shafeeqd959** [comment]
@vkalta , please resolve the comments with solution and reply for the questions. cc : @netrajpatel

File: packages/contentstack-import/src/lib/import/custom-roles.js (hunk 1 above)
Comment (vkalta) [reply packages/contentstack-import/src/lib/import/custom-roles.js:52]: @shafeeqd959 There is no loop in mapper object. Looping is done for keys in customRoles object.

**Comment by vkalta** [comment]
@shafeeqd959 I have made the changes for moving custom-roles module just before workflows.
cc: @netrajpatel

**Review by netrajpatel — APPROVED** [review]

**Pull request merged**